lib/js_browser.mli

```diff
@@ -394,6 +394,22 @@ module XHR: sig
   val set_onreadystatechange: t -> (unit -> unit) -> unit
 end
 
+module WebSocket : sig
+  type t
+
+  val create : string -> ?protocols:(string list) -> unit -> t [@@js.new "WebSocket"]
```

Comment:
I've noticed an issue once adding this optional argument. Code such as:

```ocaml
let socket = Js_browser.WebSocket.create url ()
```

results in the following runtime error:

    Uncaught TypeError: Failed to construct 'WebSocket': 1 argument required, but only 0 present.

...which I can only resolve by explicitly providing `~protocols:[]`.

Reply:
It appears as if this change to the type of `create` results in `caml_ojs_new_arr` being called with the array argument as `["ws://localhost:3000/ws"]`, where I take it it should be `["WebSocket", "ws://localhost:3000/ws"]`.

Reply:
Ah, I just saw lexifi/gen_js_api#65. Am I running into the same issue?

Reply:
Yes, very probably.  The problem is not that caml_ojs_new_arr receives the wrong array, but rather that this function call the actual constructor without the last argument from that array.